import "MsgHead.proto";    
message MsgPloughCellInfo
{
    enum GrowState {
        State_NULL    = 0;
        State_Seeding = 1;
        State_Young   = 2;
		State_Grown	  = 3;
   }
	required int32 level					= 1;
	required int32 waterPercentage			= 2;
	required GrowState state				= 3;
	required int32 manurelevel				= 4;
	required int32 seedLevel				= 5;
	required uint64 lastStateTime			= 6;
	required int32  decreaseWaterperhour	= 7;
	required int32  ID						= 8;
	required int32 waterPercentageMax	    = 9;

}

message MsgFarmInfo
{
	repeated MsgPloughCellInfo			cells = 1;
}

message MsgFarmInfoACK
{
	required MsgFarmInfo	info = 1;
}

enum enFarmErrorResult
{
	FarmError_NOTFOUNDCELL						 = 1;
	FarmError_NOTHAVESEED						 = 2;
	FarmError_CELLCANNOTSEED					 = 3;	
	FarmError_ISHAVEMANURE						 = 4;
	FarmError_GATHERNOTGROWNCELL				 = 5;
	FarmError_NOTHAVEMANURE						 = 6;
	FarmError_NOTHAVETHISSEED					 = 7;
	FarmError_ERRORSEED							 = 8;
	FarmError_THEMILLLEVELTOOHIGH				 = 9;
	FarmError_THEWATERWAYLEVELTOOHIGH			 = 10;
	FarmError_ERRORMILLLEVEL					 = 11;
	FarmError_ERRORWATERWAYLEVEL				 = 12;
	FarmError_BUILDMILLNOTHAVEENOUGHMONEY		 = 13;
	FarmError_BUILDWATERWAYNOTHAVEENOUGHMONEY	 = 14;
	FarmError_MILLISALREADYINTHISLEVEL			 = 15;
	FarmError_WATERWAYALREADYINTHISLEVEL		 = 16;
	FarmError_UKNOWN							 = 11;
}

message MsgFarmErrorACK
{
	required enFarmErrorResult en     = 1;
	required int32 cellid			  = 2;
}
message MsgSeedCellACK
{
	required int32 cellid            = 1;
	required int32 seedlevel         = 2;
}

message MsgSpreadManureACK
{
	required int32 cellid            = 1;
	required int32 manurelevel       = 2;
}

message MsgGatherPloughCellACK
{
	required int32 cellid		     = 1;
	required int32 resource		     = 2;
} 

message MsgWaterCellACK
{
	required int32 cellid			= 1;
}

message MsgBuildMillACK
{
	required int32 cellid			= 1;
	required int32 currentlevel		= 2;
	required int32 spendgold		= 3;
}

message MsgBuildWaterWayACK
{
	required int32 cellid			= 1;
	required int32 currentlevel		= 2;
	required int32 spendgold		= 3;
}